src/main/java/liquibase/ext/clickhouse/sqlgenerator/CreateDatabaseChangeLogLockTableClickHouse.java

```diff
@@ -40,20 +43,46 @@ public boolean supports(CreateDatabaseChangeLogLockTableStatement statement, Dat
     return database instanceof ClickHouseDatabase;
   }
 
+  private String generateSqlTemplate(boolean replicated) {
+    StringBuilder query = new StringBuilder();
+    query.append("CREATE TABLE IF NOT EXISTS %s.%s ");
+
+    if (replicated) query.append("ON CLUSTER '%s' ");
+
+    query.append("(");
+    query.append("ID Int64,");
+    query.append("LOCKED UInt8,");
+    query.append("LOCKGRANTED Nullable(DateTime64),");
+    query.append("LOCKEDBY Nullable(String)) ");
+
+    if (replicated) query.append("ENGINE ReplicatedMergeTree('%s','%s') ORDER BY ID");
+    else query.append("ENGINE MergeTree() ORDER BY ID");
+
+    return query.toString();
+  }
+
   @Override
   public Sql[] generateSql(
       CreateDatabaseChangeLogLockTableStatement statement,
       Database database,
       SqlGeneratorChain sqlGeneratorChain) {
+    Map<String, String> properties = ParamsLoader.getLiquibaseClickhouseProperties();
+    boolean isReplicated = properties != null;
```

Comment:
Seems like properties != null happens when we have correctly parsed the configuration file. It seems like something is missing to link it to 'isReplicated', could you please explain?

Reply:
This file in needed only when we want to use the liquibase-clickhouse in the cluster replicated mode. So if the replication properties are passed, then the plugin works in cluster replicated mode, else it works in the classic single-instance mode.